src/wallet/wallet.cpp

```diff
@@ -3437,6 +3437,45 @@ void CWallet::ListLockedCoins(std::vector<COutPoint>& vOutpts)
     }
 }
 
+
+// Note Locking Operations
+
+void CWallet::LockNote(JSOutPoint& output)
+{
+    AssertLockHeld(cs_wallet); // setLockedNotes
+    setLockedNotes.insert(output);
+}
+
+void CWallet::UnlockNote(JSOutPoint& output)
+{
+    AssertLockHeld(cs_wallet); // setLockedNotes
+    setLockedNotes.erase(output);
+}
+
+void CWallet::UnlockAllNotes()
+{
+    AssertLockHeld(cs_wallet); // setLockedNotes
+    setLockedNotes.clear();
+}
+
+bool CWallet::IsLockedNote(uint256 hash, size_t js, uint8_t n) const
+{
+    AssertLockHeld(cs_wallet); // setLockedNotes
+    JSOutPoint outpt(hash, js, n);
+
+    return (setLockedNotes.count(outpt) > 0);
+}
+
+void CWallet::ListLockedNotes(std::vector<JSOutPoint>& vOutpts)
```

Comment:
It doesn't look like ListLockedNotes is being used anywhere, this is for a future call such as listlockunspentnotes, right?

If the intent is to get a copy of setLockedNotes as a vector, you can do this:
```
std::vector v( setLockedNotes.begin(), setLockedNotes.end() );
return v;
```
Or if you want to append to an existing vector passed in as a parameter:
```
std::copy(setLockedNotes.begin(), setLockedNotes.end(), std::back_inserter(vOutpts));
```

Reply:
That's correct. It's not used yet. I mirrored the set of methods used to lock/unlock coins and this method was copied to prepare for `z_listlockunspent` and `z_lockunspent`. I don't see a reason now to have to pass in an existing vector and also don't see benefit to passing in by reference so I'll incorporate your first suggestion.